raster-foundry/raster-foundry
# Only create one intercom notifier

## Overview

This PR makes it so that we only ever create a single `IntercomNotifier`, which does enough to resolve implicits that we don't get a null pointer exception when we share with existing users.

### Checklist


### Notes

**hide whitespace changes in review** scalafmt got way too excited

There's still some kind of issue with `akka-streams` and an unsubscribed response entity that I'm working out, but this at least solves the immediate problem of "always error".

I believe this will also require some frontend changes to report the information about which email addresses had unsuccessful shares, but I don't think we know what that should look like or if that's information that we will want that we're not worried about exposing to users. Since the errors were the result of NPEs for existing users only, I don't think there are actually real errors that we needed to report.

An issue is that bad email addresses don't show up as errors -- for example, our SMTP logs show that the server was perfectly happy to pass off the message to the azavea mail server for `jsantuc@azavea.com`. I'm not sure whether there's anything we can do about that.

## Testing Instructions

- assemble api server
- bring up servers
- bring up groundwork
- go to the sharing tab of a project you're allowed to share
- put in the email address for a user who exists and share
- you shouldn't see an error

I don't know how to test the returning-all-errors part aside from trusting the `ValidatedNel` data type -- changes in eb5a35a

Closes raster-foundry/groundwork#1108
